### ISSUE TITLE
10.x.y generated files sometimes corrupt: "bad Huffman code", "premature end of data segment"

### ISSUE THREAD

**Issue by cbm755** [python-pillow/Pillow#7625]
I have jpeg files generated by Pillow 10.1.0 on different people's computers that are corrupt.

Some views choke on the files, other look fine.  Imagemagick says:
```
$ convert g1.jpg out.png
convert: Corrupt JPEG data: bad Huffman code `g1.jpg' @ warning/jpeg.c/JPEGWarningHandler/403.

$ convert g7.jpg out.png
convert: Corrupt JPEG data: premature end of data segment `g7.jpg' @ warning/jpeg.c/JPEGWarningHandler/403.
convert: Corrupt JPEG data: bad Huffman code `g7.jpg' @ warning/jpeg.c/JPEGWarningHandler/403.
```

The files were all made like this:
```python
im = PIL.Image.open("foo.png")
im.convert("RGB").save(jpgname, quality=90, optimize=True, subsampling=0)
```
   - The only non-standard thing there is the subsampling (because we have fine red lines which otherwise darken in an unpleasant way) and the `optimize=True`, which IIRC saves me about 5% or 10% file size.
   - `foo.png` is generated by Qt.
   - I'm not yet 100% confident that this isn't some filesystem troubles but I thought I'd ask first in case it looks familiar to anyone!
  

### What are your OS, Python and Pillow versions?

* OS: uncertain, I have tracked to certain users so can try to ask them what OS they run.
* Python: 3.11.6
* Pillow: 10.1.0, from `pip`.

Downstream link: https://gitlab.com/plom/plom/-/issues/3216

### Example file

Will post here once I have one that I'm allowed to share.

**Comment by radarhere** [comment]
I wonder if this is a duplicate of #7477 and #7604. Some users have found that libjpeg-turbo 3.0.0 causes some corruption. Our wheels only upgraded to that version with Pillow 10.0.1. The next release of Pillow, due out on January 2, will include libjpeg-turbo 3.0.1.

Could you test Pillow 10.0.0 and see if the problem disappears?

**Comment by cbm755** [comment]
Thanks for the tip, that looks plausible.  its a bit stochastic and I'm not sure I've seen it myself (as the files come from other people).  I just noticed it by chance today, so assumed it was new but perhaps not!  What I will first try to do is confirm that it started roughly when we bumped to `10.0.0` (July 2023).

**Comment by cbm755** [comment]
I found examples from early October (when clients would've been using 10.0.0 or 10.0.1.

No such examples 100s of files from June 30th (when we were still on 9.5.0).

I will check back in Jan after 10.2.x is out!  Maybe Feb when we have our first midterms.  Thank you for your help!

**Comment by radarhere** [comment]
Pillow 10.2.0 has now been published, with libjpeg-turbo 3.0.1 in the wheels.
